pkg/proxy/namespace_round_tripper.go

```diff
@@ -192,6 +191,10 @@ func (c *CustomNamespaceRoundTripper) addRegionLabelSelectorInPath(req *http.Req
 				                              (i)       (i + 1)   (i + 2)   (i + 3)
 			*/
 			if len(parts) > i+3 && parts[i+3] != "" {
+				// if the request is for all clusters in namespace, don't add label selector and remove all-clusters-in-pf9-tenant-namespace from path
+				if parts[i+2] == "clusters" && parts[i+3] == "all-clusters-in-pf9-tenant-namespace" {
+					parts[i+3] = ""
```

Comment:
Can we rename the parts variable to `partsOfFQDN` or similar - refactor this indexing to make it more intuitive? Perhaps using an enum? 
`(i)       (i + 1)   (i + 2)   (i + 3)`

Reply:
An example will help - and a unit test in near future